File: TEMA2/typescript/findrandom.ts
import readline from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";
const rl = readline.createInterface({ input, output });
const { floor, random } = Math;

function rand(n: number): number[] {
    const r: number[] = [];
    for (let i = 0; i < n; i++) {
        r.push(floor(random() * 100) + 1);
    }
    return r;
}

function find(x: number, n: number): string {
    const randNum = rand(n);
    let nx = 0;
    for (const i of randNum) {
        if (i === x) nx++;
    }
    return `La lista de numeros es ${randNum}\nEl numero ${x} sale ${nx} veces`;
}

const n = Number(await rl.question("Introduzca el limite: "));
const x = Number(await rl.question("Introduzca el numero a encontrar: "));
console.log(find(x, n));

rl.close();

File: TEMA2/python/findrandom.py
from random import randint

def rand(n,r=[]):
    for i in range(n):
        r.append(randint(1,100))
    return r

def find(x,n,nx=0):
    randNum=rand(n)
    for i in randNum:
        if i==x: nx+=1
    return f'La lista de numeros es {randNum}\nEl numero {x} sale {nx} veces'

n=int(input('Introduzca el limite: '))
x=int(input('Introduzca el numero a encontrar: '))
print(find(x,n))